src/category_theory/sites/dense_subsite.lean

```diff
@@ -0,0 +1,415 @@
+  end }
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of types, where `G` is full and
+cover-dense, and `ℱ, ℱ'` are sheaves, we may obtain a natural isomorphism between presheaves.
+-/
+@[simps] noncomputable
+def presheaf_iso {ℱ ℱ' : SheafOfTypes.{v} K} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) :
+  ℱ.val ≅ ℱ'.val :=
+nat_iso.of_components (λ X, app_iso H i (unop X)) (presheaf_hom H i.hom).naturality
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of types, where `G` is full and
+cover-dense, and `ℱ, ℱ'` are sheaves, we may obtain a natural isomorphism between sheaves.
+-/
+@[simps] noncomputable
+def sheaf_iso {ℱ ℱ' : SheafOfTypes.{v} K} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) : ℱ ≅ ℱ' :=
+{ hom := (presheaf_iso H i).hom, inv := (presheaf_iso H i).inv,
+  hom_inv_id' := (presheaf_iso H i).hom_inv_id, inv_hom_id' := (presheaf_iso H i).inv_hom_id }
+
+
+end types
+open types
+
+variables {ℱ : Dᵒᵖ ⥤ A} {ℱ' : Sheaf K A}
+
+/-- (Implementation). The sheaf map given in `types.sheaf_hom` is natural in terms of `X`. -/
+@[simps] noncomputable
+def sheaf_coyoneda_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  coyoneda ⋙ (whiskering_left Dᵒᵖ A Type*).obj ℱ ⟶
+  coyoneda ⋙ (whiskering_left Dᵒᵖ A Type*).obj ℱ'.val :=
+{ app := λ X, presheaf_hom H (hom_over α (unop X)), naturality' := λ X Y f,
+  begin
+  ext U x,
+  change app_hom H (hom_over α (unop Y)) (unop U) (f.unop ≫ x) =
+    f.unop ≫ app_hom H (hom_over α (unop X)) (unop U) x,
+  symmetry,
+  apply sheaf_eq_amalgamation,
+  apply H.is_cover,
+  intros Y' f' hf',
+  change unop X ⟶ ℱ.obj (op (unop _)) at x,
+  simp only [pushforward_family, functor.comp_map, coyoneda_obj_map, hom_over_app, category.assoc],
+  congr' 1,
+  conv_lhs { rw ← hf'.some.fac },
+  simp only [← category.assoc, op_comp, functor.map_comp],
+  congr' 1,
+  refine (app_hom_restrict H (hom_over α (unop X)) hf'.some.map.op x).trans _,
+  simp
+  end }
+
+/--
+(Implementation). `sheaf_coyoneda_hom` but the order of the arguments of the functor are swapped.
+-/
+@[simps] noncomputable
+def sheaf_yoneda_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  ℱ ⋙ yoneda ⟶ ℱ'.val ⋙ yoneda :=
+begin
+  let α := sheaf_coyoneda_hom H α,
+  refine { app := _, naturality' := _ },
+  { intro U,
+    refine { app := λ X, (α.app X).app U,
+      naturality' := λ X Y f, by simpa using congr_app (α.naturality f) U } },
+  { intros U V i,
+    ext X x,
+    exact congr_fun ((α.app X).naturality i) x },
+end
+
+/--
+Given an natural transformation `G ⋙ ℱ ⟶ G ⋙ ℱ'` between presheaves of arbitrary category,
+where `G` is full and cover-dense, and `ℱ'` is a sheaf, we may obtain a natural transformation
+between presheaves.
+-/
+@[simps] noncomputable
+def sheaf_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  ℱ ⟶ ℱ'.val :=
+begin
+  have α' := sheaf_yoneda_hom H α,
+  exact { app := λ X, yoneda.preimage (α'.app X),
+          naturality' := λ X Y f, yoneda.map_injective (by simpa using α'.naturality f) }
+end
```

Comment:
I think you can do
```suggestion
have α' := sheaf_yoneda_hom H α,
{ app := λ X, yoneda.preimage (α'.app X),
  naturality' := λ X Y f, yoneda.map_injective (by simpa using α'.naturality f) }
```

Reply:
Not really, but `let ... in` works for me.